GitUI/CommandsDialogs/RevisionDiffControl.cs

```diff
@@ -306,13 +307,14 @@ private async Task SetDiffsAsync(IReadOnlyList<GitRevision> revisions)
             }
         }
 
-        public void Bind(IRevisionGridInfo revisionGridInfo, IRevisionGridUpdate revisionGridUpdate, RevisionFileTreeControl revisionFileTree, Func<string>? pathFilter, Action? refreshGitStatus)
+        public void Bind(IRevisionGridInfo revisionGridInfo, IRevisionGridUpdate revisionGridUpdate, RevisionFileTreeControl revisionFileTree, Func<string>? pathFilter, Action? refreshGitStatus, IRunScript scriptRunner)
```

Comment:
The purpose of `IScriptsRunner` is to run the script, and it is a replacement of `ScriptRunner`. The purpose of `IRunScript` is to refresh the UI after a command has completed, a purpose which is not (and should not be) served by `IScriptsRunner`. Case in point: `RevisionGridControl` still extends `IRunScript`. There is, as of yet, no replacement for it.

Reply:
> Case in point: `RevisionGridControl` still extends `IRunScript`. There is, as of yet, no replacement for it.

This is correct, and this is the fundamental task that needs to be solved first - work out how to support and execute scripts outside the RevisionGrid.

Currently, the scripts are assumed to be run in the context of the `RevisionGrid`. Hence, it's expected to be available.
We have `IScriptHostControl` abstraction (introduced in #7950) which purpose is to extend the script support to other controls - e.g., it allows executing user scripts in the left panel. The same approach should work for the FileTree and the RevisionDiff controls as well. The interface can/should be extended as necessary.

I think this method's signature must change taking `IScriptHostControl` instead of `RevisionGridControl`:
https://github.com/gitextensions/gitextensions/blob/4eae18437d306aeb8ca676b78b58a1494c5db8b0/GitUI/ScriptsEngine/ScriptsManager.ScriptRunner.cs#L18-L19
Consequently, each control providing the script menu would implement `IScriptHostControl` and then pass itself to the script runner.

Here's a very [old prototype](https://github.com/gitextensions/gitextensions/compare/master...RussKie:gitextensions:User_scripts_in_FileTree) of mine (that served as a foundation for #7950) which attempted to add the script support to `RevisionFileTreeControl`.

Reply:
`IScriptHostControl` is currently a bit different, because it's used to get the current revision details. Something that, incidentally, would not be possible in the commit window. More importantly, since it doesn't wrap script running, the only way it could handle a UI refresh, is if it implemented a refresh callback, and it would become the duty of `GitUI.ScriptsEngine.ScriptsManager.ScriptRunner.RunScript` to call it.

This gives me an idea, though. It's possible to unify `IScriptHostControl`, `IRunScript.IFileListSource`, and `IRunScript` into a single interface `IScriptHostControl` that would handle all of the above. It would simply return `null` or be a no-op in cases where it's not relevant (e.g. getting the file list when running a script on the revision grid).

Then, methods like `RevisionDiffControl.Bind` can take `n `IScriptHostControl` instead of `IRunScript`, and forward any methods it doesn't handle.

I can do it, but I'm not sure what your opinion is regarding a PR that changes a bunch of things at once.

Reply:
P.S. I do want scripts in the sidebar too. But it has to be done correctly, and would be material for a separate PR. I have a script that runs `git branch -rd`, and currently, since it can only be run in the revision grid, it requires me to pick a remote branch from a tiny dialog, if I have multiple remote branches on the same revision. Being able to run it directly on a remote branch instead would be pretty great. But it would require modifying `IScriptHostControl` to be able to return a specific branch, instead of a revision.